wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/PluginFQNParser.java

```diff
@@ -42,8 +43,8 @@
 public class PluginFQNParser {
 
   private static final String INCORRECT_PLUGIN_FORMAT_TEMPLATE =
-      "Plugin '%s' has incorrect format. Should be: 'registryURL/publisher/name/version' or 'publisher/name/version'";
-  private static final String REGISTRY_PATTERN = "https?://[-./\\w]+(:[0-9]+)?(/[-./\\w]+)?";
+      "Plugin '%s' has incorrect format. Should be: 'registryURL#publisher/name/version' or 'publisher/name/version' or `referenceURL`";
```

Comment:
using `#` instead of `/` is breaking change, isn't it? Can we avoid it?

Reply:
I think no. We need to distinquish between `reference` which potentially can be any URL from `registry` + `id` which should the processed in another way.
BTW it is only changing our internal representation of the WS attributes, why it can hurt anyone ?

Reply:
I think yes, or you prepared a migration script to replace `/` -> `#` in existing workspaces?

Reply:
no i meant we can't avoid this. About existing workspaces you're right, they may be affected. But i have no idea yet how to prevent that physically.